compiler/damlc/tests/src/DA/Test/GenerateSimpleDalf.hs

```diff
@@ -58,7 +58,7 @@ main = do
             }
     let chc = TemplateChoice
             { chcLocation = Nothing
-            , chcName = ChoiceName "NotChoice"
+            , chcName = ChoiceName "Choice"
```

Comment:
This hits exactly the case that I mentioned above where the choice name and type constructor name don’t match. Let’s please keep this and test that we emit a somewhat usable error message instead of simply removing it.

Reply:
If the related type of a choice can be anything, and we support that, the why would this case be an error?

Reply:
reinstated to name the choice as "NotChoice" in the test (which works now because the code no longer assumes the choice datatype is named the same as the choice)

Reply:
As I said elsewhere, I would like us to emit an error if the choice name and the type constructor name don’t match. Having  `exercise cid Choice` call a choice called `NotChoice` is really confusing.

Reply:
I'm confused as to where you think this error check should go.

First, we should decide it this name-match really is a well-formed condition of DALF, or not.

If it isn't, then a mismatched name is confusing I agree, but not an error to be detected, and especially not when importing existing DALF via cross-sdk.

If it is an explicit well-formed condition on DALF, then perhaps the compiler should assert this condition just like it does for the well-typed condition.  And probably we shouldn't generate testing DALF which is not well-formed.